### ISSUE TITLE
xajax not initialized correctly

### ISSUE THREAD

**Issue by jgoldshlag** [evekb/evedev-kb#41]
I kept getting this error when running cron jobs:
xajax Error: xajax failed to automatically identify your Request URI.Please set the Request URI explicitly when you instantiate the xajax object.

I think the issue is that you are not passing in a URI when creating the xajax global. The fix seems to be moving the code in common/xajax/xajax.php that computes the URI up a little earlier and passing that into the constructor, like this:

```
<?php
/**
 * @package EDK
 */

require_once('common/xajax/xajax_core/xajax.inc.php');
$uri = html_entity_decode(edkURI::build(edkURI::parseURI()));
if(strpos($uri, "?") === false) $uri .= "?xajax=1";
else $uri .= "&xajax=1";
$xajax = new xajax($uri);
```

Doing this seems to fix things for me at least!

**Comment by tehfiend** [comment]
I fixed this by simply changing:
$xajax = new xajax();
to:
$xajax = new xajax('http://localhost/');
on line 7 in  ./common/xajax/xajax.php.

**Comment by Salvoxia** [comment]
Hi guys,

sorry for that!
This has been fixed with our hotfix release 4.2.12.0 (https://github.com/evekb/evedev-kb/commit/af1c61859371b0575c340d15f2099d0296cc471c).

Regards,
Salvoxia
